connectors/sync_job_runner.py

```diff
@@ -182,3 +186,22 @@ async def prepare_docs(self):
             doc["_reduce_whitespace"] = self.sync_job.pipeline["reduce_whitespace"]
             doc["_run_ml_inference"] = self.sync_job.pipeline["run_ml_inference"]
             yield doc, lazy_download
+
+    async def check_job(self):
```

Comment:
Method name/signature are confusing to me. `check_X` implies that it should either return something or raise an exception, but instead it has a side-effect of writing ingestion stats to DB.

Would breaking it into two methods like  `should_update_ingestion_stats` and `update_ingestion_stats` be better?

Reply:
Also this function does not seem to be tested, can you add tests to it?

Reply:
Will move the periodic function to a separate class, and rename the method to `update_ingestion_stats`.

>Also this function does not seem to be tested, can you add tests to it?

Reporting meta is tested in `test_sync_job_runner.py`. Do you mean the `periodically` part? No, it's not tested. Will test it in the new class.

Reply:
Add a periodical checker in https://github.com/elastic/connectors-python/pull/515/commits/f1e21232c3154ccd5afb0a889038328dab49a3a8 and https://github.com/elastic/connectors-python/pull/515/commits/f9fb26a08ac6f6d5df1e0318ee798b1c60f338d2